src/generateResponseType.js

```diff
@@ -392,23 +422,35 @@ export const unionOrInterfaceToFlow = (
                     selectedAttributes.map(
                         (attrs) =>
                             // eslint-disable-next-line flowtype-errors/uncovered
-                            ((attrs[
+                            ((attrs.attributes[
                                 hasTypeName
                             ]: any): BabelNodeObjectTypeProperty).value,
                     ),
                 ),
             );
         }
-        return sortedObjectTypeAnnotation(sharedAttributes);
+        return sortedObjectTypeAnnotation(config, sharedAttributes);
     }
     if (selectedAttributes.length === 1) {
-        return sortedObjectTypeAnnotation(selectedAttributes[0]);
+        return sortedObjectTypeAnnotation(
+            config,
+            selectedAttributes[0].attributes,
+        );
     }
-    return babelTypes.unionTypeAnnotation(
-        selectedAttributes.map((properties) =>
-            sortedObjectTypeAnnotation(properties),
+    const result = babelTypes.unionTypeAnnotation(
+        selectedAttributes.map(({typeName, attributes}) =>
+            sortedObjectTypeAnnotation(
+                {...config, path: config.path.concat([typeName])},
+                attributes,
+            ),
```

Comment:
I'm not super familiar with union types in the context of GraphQL.  Why do we need to modify the `path` in the `config` when creating union types?

Reply:
Ah, so the sub-type-name that gqlflow creates (and I'm imitating) in the case of unioins is e.g. 
```
query SomeQuery {
   someAttribute {
      ... on Human { id }
      ... on Droid { arms }
  }
}
```

```
type SomeQuery_someAttribute_Human = {id: string}
type SomeQuery_someAttribute_Droid = {arms: number}
````

So we need to add the `typeName` to the path when traversing into a union.

Reply:
This is a great example.  I'm not sure the best place for it, but I think it would help if there was a comment in the code explaining this.

Reply:
ok I'll pop it in a comment